tango/common/from_params.py

```diff
@@ -337,6 +339,12 @@ def construct_arg(
                     popped_params = Params({"type": popped_params})
             elif isinstance(popped_params, dict):
                 popped_params = Params(popped_params)
+            elif not isinstance(popped_params, Params):
+                raise TypeError(
+                    f"Expected a `Params` object, found `{popped_params}` instead while constructing "
+                    f"parameter '{argument_name}' for `{class_name}`"
+                )
```

Comment:
This also wasn't necessary for the fix, but makes `FromParams` more robust since it will catch errors sooner in the stack and print more useful information.